python/helpers/docker.py

```diff
@@ -19,69 +18,80 @@ def __init__(self, image: str, name: str, ports: Optional[Dict[str, int]] = None
+                print(f"Successfully reconnected to container: {self.name}")
             elif not self.container:
-                print(f"No container found. Starting a new one.")
+                print("No container found. Starting a new one.")
                 self.start_container()
+        except docker.errors.APIError as api_error:
+            print(f"Failed to reconnect: Docker API error: {api_error}")
         except Exception as e:
-            print(f"Failed to reconnect to the container: {e}")
+            print(f"Failed to reconnect to container: {e}")
 
     def start_container(self) -> None:
         try:
             existing_container = self.client.containers.get(self.name)
-        except NotFound:
-            existing_container = None
-
-        if existing_container:
             if existing_container.status != 'running':
-                print(f"Starting existing container: {self.name} for safe code execution...")
+                print(f"Starting existing container: {self.name}")
                 existing_container.start()
                 self.container = existing_container
-                time.sleep(5)  # This helps to get SSH ready
+                time.sleep(2)
             else:
                 self.container = existing_container
-                print(f"Container with name '{self.name}' is already running with ID: {existing_container.id}")
-        else:
-            print(f"Initializing docker container {self.name} for safe code execution...")
-            self.container = self.client.containers.run(
-                self.image,
-                detach=True,
-                ports=self.ports,
-                name=self.name,
-                volumes=self.volumes,
-            )
-            atexit.register(self.cleanup_container)
-            print(f"Started container with ID: {self.container.id}")
-            time.sleep(10)  # This helps to get SSH ready
-            self.start_health_check_thread()
+                print(f"Container {self.name} is already running.")
+        except NotFound:
+            try:
+                print(f"Creating new container: {self.name}")
+                self.container = self.client.containers.run(
+                    self.image,
+                    detach=True,
+                    ports=self.ports,
+                    name=self.name,
+                    volumes=self.volumes,
+                )
+                print(f"Started new container with ID: {self.container.id}")
+            except docker.errors.ImageNotFound as inf:
+                print(f"Image not found: {inf}")
+            except docker.errors.APIError as api_error:
+                print(f"Docker API error: {api_error}")
+            except Exception as e:
+                print(f"Unexpected error when starting container: {e}")
+            finally:
+                atexit.register(self.cleanup_container)
+                time.sleep(5)
+                self.start_health_check_thread()
```

Comment:
I'd opt to break out to variable instead for readability, just like in the original